src/orient/rational-calc.orient

```diff
@@ -0,0 +1,704 @@
+    describe(post_snark_time_parallel, "Time to generate a PoSt SNARK in parallel", secs)
+    describe(post_proof_gen, "Time to generate a PoSt Proof (including SNARKs, generating witnesses and reading from storage the challenged leaves)", secs)
+    describe(post_proof_gen_parallel, "Time to generate a PoSt Proof (including SNARKs, generating witnesses and reading from storage the challenged leaves) in parallel", secs)
+    describe(post_inclusions_time, "Time to generate merkle tree inclusion proofs for PoSt (assuming all data in memory)", secs)
+    describe(post_inclusions_time_parallel, "Time to regenerate and to read the merkle tree inclusion proofs for the challenged nodes from memory in parallel", secs)
+    describe(post_ticket_gen, "Time to generate a ticket", secs)
+
+    post_time = post_proof_gen
+    post_time_parallel = post_proof_gen_parallel
+    post_proof_gen = post_snark_time + post_inclusions_time + post_ticket_gen + post_data_access 
+    // TODO: this should be max(), not sum
+    post_proof_gen_parallel = post_snark_time_parallel + post_inclusions_time_parallel + post_data_access_parallel + post_ticket_gen
+
+    // This only accounts for generating the merkle trees
+    Ticket:
+      describe(ticket_hash, "Time to compute the hash of a node for ticket generation in PoSt", secs)
+      post_ticket_gen = ticket_hash * post_challenges
+
+    InclusionProofs:
+      Caching:
+        describe(post_mtree_cached, "Size of the cached parts of the replica merkle tree", bytes)
+        describe(post_mtree_cached_hashnodes, "Number of intermediary hash nodes cached of the replica merkle tree")
+        describe(post_mtree_cached_gib, "POST_MTREE_CACHED in GiB")
+        describe(post_mtree_overhead, "Fraction of extra storage a miner will have to use to cache the replica merkle tree")
+        describe(post_mtree_layers_cached, "Number of layers of the replica merkle tree being cached")
+        describe(post_mtree_layers_deleted, "Number of layers of the replica merkle tree not being cached")
+
+        post_mtree_cached = post_mtree_cached_hashnodes * node_size
+        post_mtree_cached_gib = post_mtree_cached / b_in_gib 
+        post_mtree_overhead = post_mtree_cached / sector_size
+        post_mtree_cached_hashnodes = expt(2, post_mtree_layers_cached) - 1
+        post_mtree_layers_deleted = commr_tree_depth - post_mtree_layers_cached
+
+      Generation:
+        describe(post_mtree_regen_hashnodes, "Number of intermediary nodes to be regenerated for a single replica merkle tree inclusion proof in PoSt")
+        describe(post_mtree_regen_nodes, "Number of leaf nodes to be access for a single replica merkle tree inclusion proof in PoSt")
+        describe(post_inclusion_time, "Time to generate a Merkle tree inclusion proof for PoSt (assuming all data in memory)", secs)
+
+        post_mtree_regen_hashnodes = expt(2, post_mtree_layers_deleted + 1) - 1 // TODO check if -1 should be removed
```

Comment:
Why `+1` in the exponentiation?

Reply:
As per @porcuquine explanation, the +1 in the exponent includes the layer on top of the deleted ones (root node) which will also have to be regenerated.